csrc/preseg_passes/make_resharding_contiguous.cpp

```diff
@@ -47,12 +47,14 @@ bool validateMeshes(Fusion* fusion) {
 // the split exprs.
 // 4. Sets the allocation domain to be the same as the loop domain with the
 // computed contiguity. This preserves both the sharding and any stride order.
+// 5. For non-resharding expressions, it moves the DIDx to the front of the
+// loop/allocation domain.
 // Note: Ideally, the loop domain can follow the logical domain and the
 // allocation domain can follow the stride order specified/inferred. However, we
 // currently require loop domain to be the same as allocation domain. This
 // behavior will be modified in the future with allocation and loop domain being
 // propagated independently.
-void setLoopAndAllocationDomain(TensorView* tv) {
+void setLoopAndAllocationDomain(TensorView* tv, bool is_resharding) {
```

Comment:
(I'm trying to avoid passing in redundant information here because it could run out-of-sync and become a nightmare to debug.)

Can is_resharding be computed from `tv->definition()` instead?

Reply:
This might not make sense. Apparently, `tv` can be the input of a resharding expression.

Reply:
We can pass the expression here but this avoids computing `isResharding` twice.